src/fheroes2/agg/agg_image.cpp

```diff
@@ -4247,7 +4247,12 @@ namespace
         case ICN::OBJNARTI:
             LoadOriginalICN( id );
             if ( _icnVsSprite[id].size() == 206 ) {
-                // If we have the Price of Loyalty assets we make a map sprite for the Magic Book artifact.
+                // These are the Price of Loyalty assets.
+
+                // Spell Scroll has an invalid offset by X axis.
+                _icnVsSprite[id][173].setPosition( ( 32 - _icnVsSprite[id][173].width() ) / 2, _icnVsSprite[id][173].y() );
```

Comment:
Hi, @ihhub, the original sprite width is 21 pixels so we have ( 32 - 21 ) / 2 = 5 pixels to shift this sprite without a need to move shadow to the left part sprite. Now we can move it 3 pixels to the left without worrying. :)

![изображение](https://github.com/user-attachments/assets/dbbaa54e-2861-4877-bc62-3cbe23c06e49)

```suggestion
                _icnVsSprite[id][173].setPosition( ( 32 - _icnVsSprite[id][173].width() ) / 2 - 3, _icnVsSprite[id][173].y() );
```

Reply:
You are absolutely right!